### PR TITLE
Update to  RealCircularGroup

### LINKED ISSUE #1365: Upgrade to RealCircleGroup
Circular is not currently working and tests are skipped until we upgrade to `CircleGroup`.

@dehann, we need https://github.com/JuliaManifolds/Manifolds.jl/issues/415 to continue.

### PR DESCRIPTION
**Pull request by Affie**
I changed to RealCircularGroup as it was the closest to what we had and it works.
close #1365

This PR also looks at #150, but in IIF to use <:InferenceVariable

### CODE REVIEW

File: src/Variables/DefaultVariables.jl
Comment (dehann) [inline src/Variables/DefaultVariables.jl:9]: yeah, I think we should just ditch the old tuples from IIF v0.25 entirely.  Doesn't block this PR though, must just remember to do before release.

File: src/services/EvalFactor.jl
Comment (dehann) [inline src/services/EvalFactor.jl:416]: thanks for fixing this and other use in IIF.

**Review by dehann — APPROVED** [review]
Code read looks good to me, I just restarted the test.  I'd like to make sure the IIF tests pass on this PR before we merge.

**Comment by dehann** [comment]
oops, this file keeps failing again and again:
`test/testExpXstroke.jl`

we must get this fixed first before we can merge

File: src/ManifoldSampling.jl
Comment (Affie) [inline src/ManifoldSampling.jl:52]: Just a breadcrumb, new numerical failure might stem from this change. I can’t test now though.

Comment (Affie) [reply src/ManifoldSampling.jl:52]: Yes. This is the issue. MKD is not calculated around the mean but identity element, so should use `identity_element`
I’ll fix if i get proper signal.

File: src/services/EvalFactor.jl
Comment (Affie) [reply src/services/EvalFactor.jl:416]: Maybe a better verb-noun is `calcSpreadBasic`

**Pull request merged**
